environment.go

```diff
@@ -284,10 +284,26 @@ func (e *envState) initSettings(path string) error {
 }
 
 func (e *envState) initDB(ctx context.Context, settings DBSettings) error {
+
 	opts := options.Client().ApplyURI(settings.Url).SetWriteConcern(settings.WriteConcernSettings.Resolve()).
 		SetReadConcern(settings.ReadConcernSettings.Resolve()).
 		SetConnectTimeout(5 * time.Second).SetMonitor(apm.NewLoggingMonitor(ctx, time.Minute, apm.NewBasicMonitor(nil)).DriverAPM())
 
+	if settings.HasAuth() {
+		ymlUser, ymlPwd, err := settings.GetAuth()
+		if err != nil {
+			return errors.Wrap(err, "problem getting auth from yaml authfile")
+		}
+		if ymlUser != "" {
+			credential := options.Credential{
+				Username: ymlUser,
+				Password: ymlPwd,
+			}
+
+			opts.SetAuth(credential)
```

Comment:
How does this differ from `clientOptions.SetAuth(credential)` above?

Reply:
They do not differ in function or intent, only in some contextual details. For example, I use GetAuthFromYAML(envAuth) instead of loading things through operations/flags.go and using settings.GetAuth(). I needed to copy most of the function to load-smoke-data.go as it is a standalone script.